server/storage/tuf_store_test.go

```diff
@@ -0,0 +1,106 @@
+package storage
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/docker/go/canonical/json"
+	"github.com/docker/notary/tuf/data"
+	"github.com/docker/notary/tuf/testutils"
+	"github.com/jinzhu/gorm"
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/stretchr/testify/require"
+)
+
+// SetupTUFSQLite creates a sqlite database for testing, wrapped by a TUFMetaStorage
+func SetupTUFSQLite(t *testing.T, dbDir string) (*gorm.DB, *TUFMetaStorage) {
+	dbStore, err := NewSQLStorage("sqlite3", dbDir+"test_db")
+	require.NoError(t, err)
+
+	consistentDBStore := NewTUFMetaStorage(dbStore)
+
+	embeddedDB := dbStore.DB
+	// Create the DB tables
+	err = CreateTUFTable(embeddedDB)
+	require.NoError(t, err)
+
+	err = CreateKeyTable(embeddedDB)
+	require.NoError(t, err)
+
+	// verify that the tables are empty
+	var count int
+	for _, model := range [2]interface{}{&TUFFile{}, &Key{}} {
+		query := embeddedDB.Model(model).Count(&count)
+		require.NoError(t, query.Error)
+		require.Equal(t, 0, count)
+	}
+	return &embeddedDB, consistentDBStore
+}
+
+// TestTUFSQLGetCurrent asserts that GetCurrent walks from the current timestamp metadata
+// to the snapshot specified in the checksum, to potentially other role metadata by checksum
+func TestTUFSQLGetCurrent(t *testing.T) {
+	tempBaseDir, err := ioutil.TempDir("", "notary-test-")
+	gormDB, tufDBStore := SetupTUFSQLite(t, tempBaseDir)
+	defer os.RemoveAll(tempBaseDir)
+	defer gormDB.Close()
+
+	ConsistentEmptyGetCurrentTest(t, tufDBStore)
+
+	// put an arbitrary piece of data in the database
```

Comment:
nit: might be good to mention the metadata is not entirely arbitrary, since it's root metadata to specifically test the inability to get to it using a consistent walk